### PR TITLE
[redis-cache] Introduce hard AnitAffinity and promote initContianers to main spec

### PR DESCRIPTION
**Pull request by dhilipkumars**
This PR introduces hard anti affinity and updates initContianrs to pod spec from annotation

### CODE REVIEW

**Comment by dhilipkumars** [comment]
cc: @prydonius could you PTAL, very small PR.

**Comment by dhilipkumars** [comment]
@prydonius 
A quick question if i run the command locally it works also in the CI 2 out of 3 replicas come up, given little more time the the chart would be up and running. 
```
W0807 16:17:43.347] + sleep 10
I0807 16:17:53.249] ERROR: Some containers failed to reach the ready state
W0807 16:17:53.350] + '[' 12 -lt 12 ']'
W0807 16:17:53.350] + '[' 1 -eq 0 ']'
W0807 16:17:53.351] + echo 'ERROR: Some containers failed to reach the ready state'
```
The verification script exhausted all its 12 re-tries, before all the three replicas are up.  If it retries a few more times it might come up. 

This is probably because this a statefulset + initContainers so containers come one after the other.

**Comment by prydonius** [comment]
@dhilipkumars yeah I guess this just needs a bit more time. @viglesiasce is there a way to increase the sleep time for specific charts in the CI? What do we do for Spinnaker?

**Comment by dhilipkumars** [comment]
@prydonius FYI, now the e2e passes.  PTAL

File: incubator/redis-cache/values.yaml [hunk 1]
```diff
@@ -3,6 +3,7 @@
 # Declare variables to be passed into your templates.
 
 ##Statefulset related configurations ss.yaml
+AntiAffinity : "hard"
```
Comment (unguiculus) [inline incubator/redis-cache/values.yaml]: Should start with a lower case letter.

Comment (dhilipkumars) [reply incubator/redis-cache/values.yaml]: agreed

**Comment by dhilipkumars** [comment]
@unguiculus & @prydonius could you PTAL now.

**Comment by dhilipkumars** [comment]
@prydonius now it passes `e2e`

**Pull request merged**